packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -855,18 +860,18 @@ class _TextSelectionHandleOverlayState
             dragStartBehavior: widget.dragStartBehavior,
             onPanStart: _handleDragStart,
             onPanUpdate: _handleDragUpdate,
-            onTap: _handleTap,
             child: Padding(
               padding: EdgeInsets.only(
                 left: padding.left,
                 top: padding.top,
                 right: padding.right,
                 bottom: padding.bottom,
               ),
-              child: widget.selectionControls!.buildHandle(
+              child: widget.selectionControls.buildHandle(
```

Comment:
is it possible we wraps the widget from the buildhandle with the gesturedetector here? It feels weird that we ask the widget to build the gestureDetector.

Reply:
It's the selection control of the user's choice that builds the handle right? The selection control decides the style of the handles and the types of interactions it suppoorts.

Reply:
It's a bit subtle that `onTap` should only work on the "knob" , while drag gestures should work on both the knob and the rectangle attached to the knob.

Reply:
That is an unfortunate how onSelectionHandleTapped behave differently based on the implementation of TextSelectionControls. 

I am ok with this approach if we can deprecate the onSelectionHandleTapped at some point....